Framework/Algorithms/inc/MantidAlgorithms/ReflectometryReductionOneAuto3.h

```diff
@@ -0,0 +1,121 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_ALGORITHMS_REFLECTOMETRYREDUCTIONONEAUTO3_H_
+#define MANTID_ALGORITHMS_REFLECTOMETRYREDUCTIONONEAUTO3_H_
+
+#include "MantidAPI/WorkspaceGroup_fwd.h"
+#include "ReflectometryWorkflowBase2.h"
+
+#include <boost/optional.hpp>
+
+namespace Mantid {
+namespace Algorithms {
+
+/** ReflectometryReductionOneAuto3 : Algorithm to run ReflectometryReductionOne,
+attempting to pick instrument parameters for missing properties. Version 3.
+*/
+class DLLExport ReflectometryReductionOneAuto3
+    : public ReflectometryWorkflowBase2 {
+public:
+  const std::string name() const override;
+  int version() const override;
+  const std::vector<std::string> seeAlso() const override {
+    return {"ReflectometryReductionOne"};
+  }
+  const std::string category() const override;
+  const std::string summary() const override;
+
+  /// Validate inputs
+  std::map<std::string, std::string> validateInputs() override;
+
+  /// For (multiperiod) workspace groups
+  bool checkGroups() override;
+  bool processGroups() override;
+
+private:
+  // Utility class to store output workspace names
+  struct WorkspaceNames {
+    std::string iVsQ;
+    std::string iVsQBinned;
+    std::string iVsLam;
+  };
+
+  class RebinParams {
```

Comment:
This class is rather large for what could just be a struct with two member functions (i.e. `hasQStep()` and `asVector()`).

Can this be replaced with a `struct`?

Reply:
This is probably worth doing but unrelated to the changes I made so I think it should be left to a future PR.